src/api/app/models/staging/staging_project.rb

```diff
@@ -225,5 +227,14 @@ def update_staging_workflow_on_backend
     def add_managers_group
       assign_managers_group(staging_workflow.managers_group)
     end
+
+    def create_project_log_entry
+      ProjectLogEntry.create!(
+        project: self,
+        user_name: User.current.login,
+        event_type: :staging_project_created,
```

Comment:
This is annoying in the specs when we need to create a few staging projects, but don't necessarily want to login (when testing _index_ actions for example)... It fails because `User.current` is `nil`

Reply:
You can add a trait to the factory that adds skip_create. But if you also want the staging project in the backend, you better login in before.

Reply:
If possible, I would like to avoid skipping the callbacks. Let's see what the others think.

Reply:
Solution 1. wouldn't work in other cases where `User.current` is needed. That's probably why we go with solution 2. in most of such cases.

The underlying problem is that we still have authorization code in some models. Until we solved this I would go with solution 2.